### DISCUSSION TITLE
SR Grant Size

### DISCUSSION THREAD

**Discussion by improperaffo** [srsran/srsRAN_Project#999] [General Help] [answered]
Dear all,
investigating the Scheduling Request (SR) and Buffer Status Report (BSR) dynamics, we noticed that the initial grant allocated after receiving an SR is determined by the following line:
```
static constexpr unsigned SR_GRANT_BYTES = 512;
```
Reading through the TS 138 321, we could not find any reference to any specific grant size to allocate after the SR reception. 

Can anybody provide some insights on how this value was chosen? 

Happy New Year,
Raphael

**Comment by frankist** [comment, marked as answer]
The TS does not define how many bytes should be granted for a detected SR, so this constant of 512 bytes is implementation-defined. 

All we know is that the chosen value should be high enough to avoid high latencies (by avoiding extra UL grants) and low enough to avoid inefficiencies (by avoiding MAC padding). We picked this value of 512 bytes because we guessed it would strike a good balance between the two.  However, I am open to suggestions for better values.

**Reply by improperaffo** [reply]
Thank you Francisco. That's what we thought.

If we find that other values work better, we will surely let you know. For now, we agree that 512 seems a good balance.
